src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationLoader.cs

```diff
@@ -116,9 +116,9 @@ private void OnRequiredAssemblyDetected()
             }
         }
 
-        private string GetAssemblyName(Assembly assembly)
+        private string? GetAssemblyName(Assembly assembly)
         {
-            return assembly.FullName.Split(new[] { ',' }, count: 2)[0];
+            return assembly.FullName?.Split(new[] { ',' }, count: 2)[0];
```

Comment:
I think it could be `!` instead?

Reply:
You are probably right. It is too defensive programming, but as long as we use it only for the comparison, we can accept null as return.